--- protobuf-java-compiler/src/main/java/com/rubberjam/protobuf/compiler/Parser.java
+++ protobuf-java-compiler/src/main/java/com/rubberjam/protobuf/compiler/Parser.java
@@ -758,11 +758,11 @@
                   field.setDefaultValue(input.current().text);
                   input.next();
               } else {
-                  StringBuilder sb = new StringBuilder();
-                  if (consumeNumber(sb)) {
-                      field.setDefaultValue(sb.toString());
-                  } else if (lookingAtType(Tokenizer.TokenType.IDENTIFIER)) {
+                  if (lookingAtType(Tokenizer.TokenType.IDENTIFIER) && !lookingAt("inf") && !lookingAt("nan")) {
                       field.setDefaultValue(input.current().text);
                       input.next();
+                  } else {
+                      StringBuilder sb = new StringBuilder();
+                      if (consumeNumber(sb)) {
+                          field.setDefaultValue(sb.toString());
+                      }
                   }
               }
           } else {
