--- protobuf-java-compiler/src/main/java/com/rubberjam/protobuf/compiler/java/full/EnumGenerator.java
+++ protobuf-java-compiler/src/main/java/com/rubberjam/protobuf/compiler/java/full/EnumGenerator.java
@@ -131,8 +131,9 @@
       vars.put("canonical_name", alias.canonicalValue.getName());
       printer.print("/**\n");
       DocComment.writeDocCommentBody(printer, alias.value, context.getOptions(), false);
-      printer.print(vars, " * <code>$name$ = " + alias.value.getNumber() + (alias.value.getOptions().getDeprecated() ? " [deprecated = true]" : "") + ";</code>\n" +
+      printer.print(vars, " * <code>$name$ = " + alias.value.getNumber() + (alias.value.getOptions().getDeprecated() ? " [deprecated = true]" : "") + ";</code>\n");
       printer.print(" */\n");
       String deprecation = alias.value.getOptions().getDeprecated() ? "@java.lang.Deprecated " : "";
       printer.print(vars,
           deprecation + "public static final $classname$ $name$ = $canonical_name$;\n");
