--- protobuf-java-compiler/src/main/java/com/rubberjam/protobuf/compiler/java/full/EnumGenerator.java
+++ protobuf-java-compiler/src/main/java/com/rubberjam/protobuf/compiler/java/full/EnumGenerator.java
@@ -128,7 +128,11 @@
       vars.put("classname", descriptor.getName());
       vars.put("name", alias.value.getName());
       vars.put("canonical_name", alias.canonicalValue.getName());
+      printer.print("/**\n");
+      DocComment.writeDocCommentBody(printer, alias.value, context.getOptions(), false);
+      printer.print(vars, " * <code>$name$ = " + alias.value.getNumber() + (alias.value.getOptions().getDeprecated() ? " [deprecated = true]" : "") + ";</code>\n");
+      printer.print(" */\n");
       printer.print(vars,
-          "public static final $classname$ $name$ = $canonical_name$;\n");
+          "public static final $classname$ $name$ = $canonical_name$;\n");
     }

     for (EnumValueDescriptor value : descriptor.getValues()) {
