--- protobuf-java-compiler/src/main/java/com/rubberjam/protobuf/compiler/java/full/EnumGenerator.java
+++ protobuf-java-compiler/src/main/java/com/rubberjam/protobuf/compiler/java/full/EnumGenerator.java
@@ -145,10 +145,10 @@
       vars.put("deprecation", value.getOptions().getDeprecated() ? "@java.lang.Deprecated " : "");

       printer.print("/**\n");
       DocComment.writeDocCommentBody(printer, value, context.getOptions(), false);
       printer.print(vars,
-          " * <code>$name$ = $number$;</code>\n" +
+          " * <code>$name$ = $number$" + (value.getOptions().getDeprecated() ? " [deprecated = true]" : "") + ";</code>\n" +
           " */\n");

       printer.print(vars,
           "$deprecation$public static final int $name$_VALUE = $number$;\n");
